data/tools/wesnoth/wmltools3.py

```diff
@@ -705,15 +778,22 @@ def scan_for_definitions(self, namespace, filename):
                                             % (here, name, defn), file=sys.stderr)
                         if name not in self.xref:
                             self.xref[name] = []
+
+                        here.lineno_end = n+1
                         self.xref[name].append(here)
                         state = States.OUTSIDE
                     elif state == States.MACRO_HEADER and line.strip():
                         if line.strip().startswith("#arg"):
                             state = States.MACRO_OPTIONAL_ARGUMENT
-                            here.optional_args.append(line.strip().split()[1])
+                            here._optional_args.append([line.strip().split()[1],""])
                         elif line.strip()[0] != "#":
                             state = States.MACRO_BODY
-                    elif state == States.MACRO_OPTIONAL_ARGUMENT and "#endarg" in line:
+                    elif state == States.MACRO_OPTIONAL_ARGUMENT and not "#endarg" in line:
+                        here._optional_args[-1][1] += line
+                    elif state == States.MACRO_OPTIONAL_ARGUMENT:
+                        end_arg_index = line.index("#endarg")
+                        here._optional_args[-1][1] += line[0:end_arg_index]
```

Comment:
It can be really confusing to have two variables called `_optional_args` and `optional_args`, with the only difference being a leading underscore. Please consider making the two names a bit more different.

Reply:
True. Do you think ``_raw_optional_args`` would be appropriate?